modules/core/include/opencv2/core/dualquaternion.hpp

```diff
@@ -0,0 +1,857 @@
+ * &= [w, x, y, z, w\_, x\_, y\_, z\_]
+ * \end{split}
+ * \end{equation}
+ * \f]
+ * where \f$r, t\f$ represents the rotation (ordinary unit quaternion) and translation (pure ordinary quaternion) respectively.
+ *
+ * A more general dual quaternions who consist of two quaternions are usually represented in the form:
+ * \f[
+ * \sigma = p + \epsilon q
+ * \f]
+ * where the introduced dual unit \f$\epsilon\f$ satisfies \f$\epsilon^2 = \epsilon^3 =...=0\f$, and \f$p, q\f$ are all quaternions.
+ *
+ * Alternatively, dual quaternions can also be interpreted as four components which are all dual numbers:
+ * \f[
+ * \sigma = \hat{q}_w + \hat{q}_xi + \hat{q}_yj + \hat{q}_zk
+ * \f]
+ * If we set \f$\hat{q}_x, \hat{q}_y\f$ and \f$\hat{q}_z\f$ equal to 0, a dual quaternion is transformed to a dual number. see normalize().
+ *
+ * If you want to create a dual quaternion, you can use:
+ *
+ * ```
+ * using namespace cv;
+ * double angle = CV_PI;
+ *
+ * // create from eight number
+ * DualQuatd dq1(1, 2, 3, 4, 5, 6, 7, 8); //p = [1,2,3,4]. q=[5,6,7,8]
+ *
+ * // create from Vec
+ * Vec<double, 8> v{1,2,3,4,5,6,7,8};
+ * DualQuatd dq_v{v};
+ *
+ * // create from two quaternion
+ * Quatd p(1, 2, 3, 4);
+ * Quatd q(5, 6, 7, 8);
+ * DualQuatd dq2 = DualQuatd::createFromQuat(p, q);
+ *
+ * // create from an angle, an axis and a translation
+ * Vec3d axis{0, 0, 1};
+ * Quatd trans{0, 3, 4, 5};
+ * DualQuatd dq3 = DualQuatd::createFromAngleAxisTrans(angle, axis, trans);
+ *
+ * // create from a linear transformation matrix R.
+ * // see createFromMat() in detail for the form of R
+ * Matx44d R = dq3.toMat();
+ * DualQuatd dq4 = DualQuatd::createFromMat(R);
+ *
+ * // create from screw parameter.
+ * Quatd qaxis{0, 0, 0, 1}; // that is [0, axis], qaxis should be normalized.
+ * Quatd t{0, 3, 4 ,5};
+ * Quatd moment = 1.0 / 2 * (t.crossProduct(qaxis) + qaxis.crossProduct(t.crossProduct(qaxis)) * std::cos(angle / 2) / std::sin(angle / 2));
+ * double d = t.dot(qaxis);
+ * DualQuatd dq5 = DualQuatd::createFromPitch(angle, d, qaxis, moment);
+ * ```
+ *
+ * A point \f$v=(x, y, z)\f$ in form of dual quaternion is \f$[1+\epsilon v]=[1,0,0,0,0,x,y,z]\f$.
+ * The transformation of a point \f$v_1\f$ to another point \f$v_1\f$ under the dual quaternion \f$\sigma\f$ is
+ * \f[
+ * 1 + \epsilon v_2 = \sigma * (1 + \epsilon v_1) * \sigma^{\star}
+ * \f]
+ * where \f$\sigma^{\star}=p^*-\epsilon q^*.\f$
+ *
+ * A line in the \f$Pl\ddot{u}cker\f$ \f$(\hat{l}, m)\f$ by the dual quaternion \f$l=\hat{l}+\epsilon m\f$.
+ * To transform a line, \f[l_2 = \sigma * l_1 * \sigma^*,\f] where \f$\sigma=r+\frac{\epsilon}{2}rt\f$ and
+ * \f$\sigma^*=p^*+\epsilon q^*\f$.
+ *
+ * To extract the Vec<double, 8> or Vec<float, 8>, see toVec();
+ *
+ * To extract the transformation matrix, see toMat();
+ *
+ * If there are two quaternions \f$q_0, q_1\f$ are needed to interpolate, you can use sclerp():
+ * ```
+ * Quatd::sclerp(q0, q1, t)
+ * ```
+ */
```

Comment:
I think there should be some explanation on how to do things that can be done by Affine3 class, like how to compose poses, how to apply a pose to a point, convert them between representations and so on.

Reply:
OK, I will add the usage of Affine3 in detail later.